libpod/container_internal_linux.go

```diff
@@ -1247,6 +1275,79 @@ func (c *Container) restore(ctx context.Context, options ContainerCheckpointOpti
 		}
 	}
 
+	if options.Pod != "" {
+		// Running in a Pod means that we have to change all namespace settings to
+		// the ones from the infrastructure container.
+		pod, err := c.runtime.LookupPod(options.Pod)
+		if err != nil {
+			return errors.Wrapf(err, "pod %q cannot be retrieved", options.Pod)
+		}
+
+		infraContainer, err := pod.InfraContainer()
+		if err != nil {
+			return errors.Wrapf(err, "cannot retrieved infra container from pod %q", options.Pod)
+		}
+
+		if err := infraContainer.syncContainer(); err != nil {
+			return errors.Wrapf(err, "Error syncing infrastructure container %s status", infraContainer.ID())
+		}
+		if infraContainer.state.State != define.ContainerStateRunning {
+			if err := infraContainer.initAndStart(ctx); err != nil {
+				return errors.Wrapf(err, "Error starting infrastructure container %s status", infraContainer.ID())
+			}
+		}
+
+		if c.config.IPCNsCtr != "" {
```

Comment:
This isn't really matching the way we do pods - the pod itself stores what namespaces it shares. I would not expect good results if you restore a container from a pod that shared the PID namespace, into a pod that does not share the PID namespace.

Reply:
Also, for all of these - you need to ensure that the container config field is set to the ID of the new pod's infra container. I think that the DB will not allow you to add a container to the DB if this isn't true, but we should make sure.

Reply:
I added checks to make sure only container are restored into Pods with the same namespace shared as the checkpointed container.